internal/controller/isbservicerollout_controller.go

```diff
@@ -277,33 +301,61 @@ func (r *ISBServiceRolloutReconciler) processExistingISBService(ctx context.Cont
 	// update our Status with the ISBService's Status
 	r.processISBServiceStatus(ctx, existingISBServiceDef, isbServiceRollout)
 
-	// if I need to update or am in the middle of an update of the ISBService, then I need to make sure all the Pipelines are pausing
-	isbServiceNeedsUpdating, isbServiceIsUpdating, err := r.isISBServiceUpdating(ctx, isbServiceRollout, existingISBServiceDef)
+	// if I am in the middle of an update of the ISBService, then I need to make sure all the Pipelines are pausing
+	_, isbServiceIsUpdating, err := r.isISBServiceUpdating(ctx, isbServiceRollout, existingISBServiceDef)
 	if err != nil {
 		return false, err
 	}
 
-	numaLogger.Debugf("isbServiceNeedsUpdating=%t, isbServiceIsUpdating=%t", isbServiceNeedsUpdating, isbServiceIsUpdating)
+	// determine if we're trying to update the ISBService spec
+	// if it's a simple change, direct apply
+	// if not, it will require PPND or Progressive
+	isbServiceNeedsToUpdate, upgradeStrategyType, err := usde.ResourceNeedsUpdating(ctx, newISBServiceDef, existingISBServiceDef)
+	if err != nil {
+		return false, err
+	}
+	numaLogger.
+		WithValues("isbserviceNeedsToUpdate", isbServiceNeedsToUpdate, "upgradeStrategyType", upgradeStrategyType).
+		Debug("Upgrade decision result")
 
 	// set the Status appropriately to "Pending" or "Deployed"
 	// if isbServiceNeedsUpdating - this means there's a mismatch between the desired ISBService spec and actual ISBService spec
 	// Note that this will be reset to "Deployed" later on if a deployment occurs
-	if isbServiceNeedsUpdating {
+	if isbServiceNeedsToUpdate {
 		isbServiceRollout.Status.MarkPending()
 	} else {
 		isbServiceRollout.Status.MarkDeployed(isbServiceRollout.Generation)
 	}
 
-	// determine the Upgrade Strategy user prefers
-	upgradeStrategy, err := usde.GetUserStrategy(ctx, isbServiceRollout.Namespace)
+	// what is the preferred strategy for this namespace?
+	userPreferredStrategy, err := usde.GetUserStrategy(ctx, newISBServiceDef.Namespace)
 	if err != nil {
 		return false, err
 	}
 
-	switch upgradeStrategy {
-	case config.PPNDStrategyID:
+	// is there currently an inProgressStrategy for the isbService? (This will override any new decision)
+	inProgressStrategy := r.inProgressStrategyMgr.getStrategy(ctx, isbServiceRollout)
+	inProgressStrategySet := (inProgressStrategy != apiv1.UpgradeStrategyNoOp)
 
-		return processChildObjectWithPPND(ctx, r.client, isbServiceRollout, r, isbServiceNeedsUpdating, isbServiceIsUpdating, func() error {
+	// if not, should we set one?
+	if !inProgressStrategySet {
+		if userPreferredStrategy == config.PPNDStrategyID {
+			if upgradeStrategyType == apiv1.UpgradeStrategyPPND {
```

Comment:
nit: I know you copied this from my redundant code in PipelineRolloutController, so my bad, but I believe the check for `userPreferredStrategy` here and below may be unnecessary as it's happening within `ResourceNeedsUpdating()`. We probably don't need to get the value for `userPreferredStrategy` above at all.